plover/gui/paper_tape.py

```diff
@@ -170,6 +198,26 @@ def display(parent, config):
         # StrokeDisplayDialog shows itself.
         StrokeDisplayDialog(parent, config)
 
+class NonModalSaveDialog(wx.FileDialog):
+    def __init__(self,parent,notes_to_save):
+        self.notes_to_save = notes_to_save
+        file_name_suggestion = None
+        counter = 0
+        while not file_name_suggestion or os.path.exists(file_name_suggestion):
+            file_name_suggestion = "steno-notes-"+time.strftime("%Y-%M-%d")+"("+str(counter)+").txt"
+            counter += 1
```

Comment:
The string generation could be made a little more readable with string formatting:

```python
        counter = 0
        while not file_name_suggestion or os.path.exists(file_name_suggestion):
            counter_string = ' (%i)' % counter if counter else ''
            file_name_suggestion = "steno-notes-%s%s.txt" % (time.strftime("%Y-%M-%d"), counter_string)
            counter += 1
```